physionet-django/console/views.py

```diff
@@ -716,6 +699,48 @@ def manage_published_project(request, project_slug, version):
          'rw_tasks': rw_tasks, 'ro_tasks': ro_tasks,
          'anonymous_url':anonymous_url, 'passphrase':passphrase})
 
+def gcp_bucket_management(request, project, user):
+    """
+    Create the database object and cloud bucket if they do not exist, and send
+    the files to the bucket.
+    """
+    is_private = True
+
+    if project.access_policy == 0:
+        is_private = False
+        group = None
+    else:
+        group = utility.get_bucket_email(project.slug, project.version)
+
+    bucket_name = utility.get_bucket_name(project.slug, project.version)
+
+    if GCP.objects.filter(bucket_name=bucket_name):
```

Comment:
Is it possible to simplify this logic, perhaps by removing the nesting and/or moving some of the content to another function? As it is now, I find the steps difficult to follow.

Reply:
I would not make sense to move to another function since this was already moved to another function, plus the function is not that long nor complicates since half the function are messages and loggers. I can "simplify the logic" by converting an if/else for a try/except.

Reply:
I'll take another look and will give it a bit more thought. It feels like a minor rearrangement could make the function easier to follow.